java/client/src/main/java/glide/api/commands/StringBaseCommands.java

```diff
@@ -74,6 +76,41 @@ public interface StringBaseCommands {
      */
     CompletableFuture<String> getdel(String key);
 
+    /**
+     * Gets the value associated with the given <code>key</code>.
+     *
+     * @since Redis 6.2.0.
+     * @see <a href="https://redis.io/docs/latest/commands/getex/">redis.io</a> for details.
+     * @param key The <code>key</code> to retrieve from the database.
+     * @return If <code>key</code> exists, return the <code>value</code> of the <code>key</code>.
+     *     Otherwise, return <code>null</code>.
+     * @example
+     *     <pre>{@code
+     * String value = client.getex("key").get();
+     * assert value.equals("value");
+     * }</pre>
+     */
+    CompletableFuture<String> getex(String key);
+
+    /**
+     * Gets the value associated with the given <code>key</code>.
+     *
+     * @since Redis 6.2.0.
+     * @see <a href="https://redis.io/docs/latest/commands/getex/">redis.io</a> for details.
+     * @param key The <code>key</code> to retrieve from the database.
+     * @param options The {@link GetExOptions} options.
+     * @return If <code>key</code> exists, return the <code>value</code> of the <code>key</code>.
+     *     Otherwise, return <code>null</code>.
+     * @example
+     *     <pre>{@code
+     * String response = client.set("key", "value").get();
+     * assert response.equals(OK);
+     * String value = client.getex("key", GetExOptions.Seconds(10L)).get();
+     * assert value.equals("value");
+     * }</pre>
+     */
+    CompletableFuture<String> getex(String key, @NonNull GetExOptions options);
```

Comment:
The rest of this file doesn't use Nonnull, looking at the other BaseCommands.java files we seem to be inconsistent with whether we include Nonnull or not in the interfaces. Regardless, I think you should either add Nonnull to the key as well (for both getex signatures) or remove the Nonnull here

Reply:
i will add back @nonnull for the key

Reply:
remove it **